### PR TITLE
Fix recipe card display format

### PR DESCRIPTION
**Pull request by WeichenXu123**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/WeichenXu123/mlflow/pull/10893?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10893/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10893
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

The security fix https://github.com/mlflow/mlflow/pull/10873 causes Recipe card display format broken.
This PR fixes it.

### How is this PR tested?

Manually test:
image

image

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by WeichenXu123** [comment]
@mlflow-automation autoformat

File: mlflow/recipes/cards/__init__.py [hunk 1]
```diff
@@ -237,11 +237,19 @@ def render_table(table, columns=None, hide_index=True):
...
+
         import pandas as pd
         from pandas.io.formats.style import Styler
 
         if not isinstance(table, Styler):
-            table = pd.DataFrame(table, columns=columns).style
+            table = pd.DataFrame(table, columns=columns)
+            # Escape specific characters in HTML to prevent
+            # javascript code injection
+            # Note that `pandas_df.style.to_html(escape=True) does not work
```
Comment (harupy) [inline mlflow/recipes/cards/__init__.py:249]: why `pandas_df.style.to_html(escape=True)` does not work? What's the issue?

Comment (WeichenXu123) [reply mlflow/recipes/cards/__init__.py:249]: I think it is a pandas library bug. We can report it.

Comment (harupy) [reply mlflow/recipes/cards/__init__.py:249]: what happens if we use `pandas_df.style.to_html(escape=True)`?

File: mlflow/recipes/cards/__init__.py [hunk 2]
```diff
@@ -290,8 +298,12 @@ def __init__(
...
         )
         self.add_tab(
-            "Stacktrace", "<div class='stacktrace-container'>{{ STACKTRACE|e }}</div>"
-        ).add_html("STACKTRACE", f'<p style="margin-top:0px"><code>{failure_traceback}</code></p>')
+            "Stacktrace",
+            (
+                "<div class='stacktrace-container'><p style='margin-top:0px'><code>"
+                "{{ STACKTRACE|e }}</code></p></div>"
+            ),
+        ).add_html("STACKTRACE", f"{failure_traceback}")
```
[the remark this answers is not in the document]
Comment (harupy) [reply mlflow/recipes/cards/__init__.py:306]: or

```suggestion
        ).add_html("STACKTRACE", str(failure_traceback))
```

if `failure_traceback` is not a string.

File: mlflow/recipes/cards/__init__.py (hunk 1 above)
Comment (WeichenXu123) [reply mlflow/recipes/cards/__init__.py:249]: It generate the same result with `pandas_df.style.to_html(escape=False)`

But `pandas_df.to_html(escape=True)` generate escaped result correctly.

So I think it is a bug.

You can try it by:

```
p1 = pd.DataFrame({'a': ['<div><script></script></div>']})

print(p1.style.to_html(escape=False))
print(p1.style.to_html(escape=True))

print(p1.to_html(escape=False))
print(p1.to_html(escape=True))
```

**Review by harupy — APPROVED** [review]
lgtm

**Pull request merged**